pycqed/measurement/waveform_control/circuit_builder.py

```diff
@@ -852,8 +855,21 @@ def sweep_n_dim(self, sweep_points, body_block=None, body_block_func=None,
 
                 segblock = self.sequential_blocks(
                         'segblock', [prep, this_body_block, final, ro])
-                seq.add(Segment(f'seg{j}', segblock.build(
-                    sweep_dicts_list=sweep_points, sweep_index_list=[j, i])))
+                seg = Segment(f'seg{j}', segblock.build(
+                    sweep_dicts_list=sweep_points, sweep_index_list=[j, i]))
+                # apply Segment sweep points
+                for dim in [0, 1]:
+                    if len(sweep_points[dim]) == 0:
+                        continue
+                    for param, vals in [
+                            [s[2], s[0]] for s in
+                            sweep_points.get_sweep_params_description(
```

Comment:
I think we should use `get_sweep_param_property('values', dim, 'all')` and `get_sweep_param_property('label', dim, 'all')` instead of s[0], s[2], no? The point of writing these was to not have to dependencies if we ever change the index of these parameters in `sweep_params_description`.

Reply:
In fact I did not event want the label, but rather the param_name. Thanks for spotting this. Should be fixed now.